File: backend/entity/Payment.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Payment struct {
	gorm.Model
	PaymentDate time.Time `json:"PaymentDate" valid:"required~PaymentDate is required"`
	Amount      float32   `json:"Amount" valid:"required~Amount is required"`
	Slip        string    `json:"Slip" gorm:"type:longtext" valid:"required~Slip is required" `
	TransactionLog []TransactionLog `gorm:"foreignKey:PaymentID"`

	UserID uint `json:"UserID" valid:"-"`
	User   User `gorm:"foreignKey:UserID"`

	PurchaseID uint     `json:"PurchaseID" valid:"-"`
	Purchase   Purchase `gorm:"foreignKey:PurchaseID"`

	MethodID      uint          `json:"MethodID" valid:"-"`
	PaymentMethod PaymentMethod `gorm:"foreignKey:MethodID"`
}

File: frontend/src/Interface/Payment.ts
import { UserInterface } from "./IUser";
import { PurchaseInterface } from "./Purchase";
import { TransactionLogInterface } from "./TransactionLog";

export interface PaymentInterface {
    ID?: number;                     
    PaymentDate: Date;               
    Amount: number;                 
    Slip: string;                   
  
    TransactionLog?: TransactionLogInterface[]; 
    UserID: number;                
    User?: UserInterface;           
  
    PurchaseID: number;             
    Purchase?: PurchaseInterface;    
  
    MethodID: number;               
    PaymentMethod?: PaymentInterface; 
  }
  